docs/APP_DEVELOPMENT.md

```diff
@@ -29,69 +28,158 @@ An app typically has the structure:
         │       └── <app-name>.js
         ├── config.json
         ├── data
+        ├── swagger-extensions.yaml
         └── index.js
 
-At the top level, the ``README.md`` file explains the purpose and usage of the application in human-readable form, whereas ``package.json`` provides machine-readable information about the package (name, version, author, license), its dependencies, and build commands.
-
-Within ``src/``, ``config.json`` describes any pre-configured ``states`` that are provided as examples (which may depend on data files in ``src/data``); ``src/index.js`` is the file that is actually run by node.js to create a server instance.
+At the top level, the `README.md` file explains the purpose and usage of the application in human-readable form, whereas `package.json` provides machine-readable information about the package (name, version, author, license), its dependencies, and build commands.
 
-Apps are partitioned into separate a ``control`` and ``view``, with shared parts placed in ``src/client/common/`` (for CSS or JavaScript functions) or ``src/client/constants`` (for JavaScript constants).
-A single ``index.html`` file is shared between the ``control`` and ``view``, but it may render different content in both cases due to the inclusion of different JavaScript files.
+Within `src/`, `config.json` describes any pre-configured `states` that are provided as examples (which may depend on data files in `src/data`); `src/index.js` is the file that is actually run by node.js to create a server instance.
+The `@ove-lib/appbase` base library exposes an API to interact with the app. If the application exposes additional API methods the `src/swagger-extensions.yaml` is used to describe them.
```

Comment:
"the `src/swagger-extensions.yaml` is used to describe them **so that [Swagger](https://swagger.io/solutions/api-documentation/) can automatically generate documentation**"